consensus/src/units/store.rs

```diff
@@ -28,6 +79,46 @@ impl<'a, H: Hasher, D: Data, KB: KeyBox> UnitStore<'a, H, D, KB> {
         }
     }
 
+    pub fn get_status(&self) -> UnitStoreStatus {
+        fn first_missing(v: Vec<Round>) -> Round {
+            let mut w = vec![false; v.len()];
+            for r in &v {
+                if let Some(f) = w.get_mut(*r as usize) {
+                    *f = true;
+                }
+            }
+            match w.into_iter().position(|x| !x) {
+                Some(r) => r as u16,
+                None => v.len() as u16,
+            }
+        }
+        let gm = self
+            .by_coord
+            .keys()
+            .map(|c| (c.creator.0, c.round))
+            .into_grouping_map();
+        let top_row: HashMap<usize, Round> = gm.clone().max();
+        let mut first_missing_rounds: Vec<(usize, Round)> = gm
+            .collect::<Vec<_>>()
```

Comment:
rewritten to use `NodeMap`, check if it's okay in the current state

Reply:
This was part of a different problem, not the `NodeMap` one. I thought about this a tad and lemme write the code I claim is better:
```
gm.
 .collect<BTreeSet<_>>()
 .into_iter()
 .filter_map(|(id, rounds)| rounds.into_iter().zip((0..)).position(|(round, index)| round != index).map(|round| (id, round)))
 .collect()
```

And then get rid of `first_missing`.

Reply:
`BTreeSet` is sorted out of the box.

Reply:
then it's slower (at least in theory...), I don't like it!
@DamianStraszak - please be the judge

Reply:
Actually I find `first_missing` easier to understand than these kind of one-liners. Also this creates a heavy `BTreeSet` only to sort stuff.

Reply:
@timorl - are you ok with this?

Reply:
How about
```
gm.
 .collect<HashSet<_>>()
 .into_iter()
 .filter_map(|(id, rounds)| (0..top_row.get(id)).position(|round| !rounds.contains(round)).map(|round| (id, round)))
 .collect()
```
?

I don't like `first_missing` mostly because such "purely algorithmic" code appearing in a codebase is usually a red flag (esp. with single letter variable names) and often hard to understand – the call itself is relatively easy to understand, but the code of the function raises questions. I would argue that while my previous attempt wasn't easy to understand the one above is about as simple as the `first_missing` version, especially without the additional `filter`.

Reply:
Ok, when it comes to efficiency all this anyway uses `into_grouping_map()` so this is not making it so much slower. I'm ok with that version.

If we wanted this to actually cause low overhead (and note that for 100 nodes this will already be quite heavy) one would need to maintain `top_row` and `first_missing` directly in the store. You can easily update in amortized `O(1)` time per inserted unit. But it's not worth it at this point.

Reply:
fixed
now one could say that `first_missing` is... missing